### PR TITLE
Feature/client side routing serialize query

### PR DESCRIPTION
**Pull request by tilt**
https://www.openproject.org/work_packages/8444

### CODE REVIEW

File: app/assets/javascripts/angular/routing.js [hunk 1]
```diff
@@ -26,27 +26,51 @@
...
 
-  $urlRouterProvider.otherwise("/wp");
+  // All unmatched routes should be handled by Rails
+  // (i.e. invoke a full-page load)
+  //
+  // a better behaviour here would be for Angular's HTML5-mode to provide an
+  // option to turn link hijacking off by default (and use a `target` attribute
+  // to turn it on for selected links).
+
+  $urlRouterProvider.otherwise(function($injector, $location) {
```
Comment (myabc) [inline app/assets/javascripts/angular/routing.js]: @tilt I'm not sure how this got back in here (perhaps a dodgy merge in 5812bc78?) In any case, this and the `run` function below should not both be present.

Comment (tilt) [inline app/assets/javascripts/angular/routing.js]: can you remove it again? Should I? There's also again a failing cuke after dev was merged back :(

Comment (myabc) [inline app/assets/javascripts/angular/routing.js]: @tilt I can remove it again.

Comment (tilt) [inline app/assets/javascripts/angular/routing.js]: thanks!

Comment (tilt) [inline app/assets/javascripts/angular/routing.js]: that postgres spec failure seems to be recurring btw

**Comment by myabc** [comment]
> this set_filter could be removed everywhere now

I'd like to, but it looks like calendars and cost reports use this parameter too.

**Pull request merged**